src/store/modules/milk/selectors.ts

```diff
@@ -0,0 +1,127 @@
+import { createSelector } from 'reselect';
+import { MilkEntry } from './types';
+import { State } from '../../types';
+
+import * as moment from 'moment';
+
+const getMilkEntries = (state: State) => state.milk.rows;
+const getCurrentFarmerUUID = (state: State) => state.currentFarmer.currentFarmerUUID;
+const radix: number = 10;
+
+/************Selectors for all milk entries (used on Home page) ***************/
+/**Selector to calculate the current days milk collection */
+export const getDaysMilkTotal = createSelector(
+  [getMilkEntries],
+  (milkEntries: MilkEntry[]) => milkEntries.reduce((sum: number, entry: MilkEntry) =>
+    inSameDay(entry.datetime) ? sum + parseInt(entry.volume, radix) : sum + 0, 0).toFixed(0));
+
+/**Selector to calculate the average daily milk collection */
+export const getAvgDaysMilkTotal = createSelector(
+  [getMilkEntries],
+  (milkEntries: MilkEntry[]) => {
+    return calculateAverage(groupBy(milkEntries.map(entry =>
+      ({ ...(entry as any), dateID: moment(entry.datetime).utc().format() })), 'dateID'));
```

Comment:
Is this the average for this day in all of history or the average for today? I feel like it should be the average for today, in which case it wouldn't be using a groupBy or anything complicated, it would just use the previous selector and a `numberOfMilkEntriesToday` selector you'd have to build

Reply:
This is the average for all days on record. My reasoning was so you'd have an idea of how it's going vs other days in the db

Reply:
Ahh k, well in that case take a look at [ramda](http://ramdajs.com/docs/#groupBy). I installed it on my branch already because I'm processing so much data in the sync service, it has a `groupBy` function that should work for what you're doing. If you use this make sure to run `yarn add --dev types/npm-ramda#dist` to install the types instead of `@types`

Reply:
You're welcome to keep using the function you wrote if the API is significantly different and hard to convert

Reply:
Lets keep using it for now and I'll explore changing it when we have some breathing room